react/Carousel.tsx

```diff
@@ -290,71 +90,255 @@ export default class Carousel extends Component<Props, State> {
     )
   }
 
-  public render() {
-    const { height, showArrows, autoplay, autoplaySpeed, showDots } = this.props
-    const { currentSlide } = this.state
-    if (!this.props.banners.length) {
-      return null
-    }
+  const banners: BannerProps[] = bannersProp.filter(
+    banner => banner && (banner.mobileImage || banner.image)
+  )
 
-    const banners: BannerProps[] = this.props.banners.filter(
-      banner => banner && (banner.mobileImage || banner.image)
-    )
+  const handleNextSlide = () => {
+    const nextSlide = ((currentSlide + 1 - PER_PAGE) % banners.length) + PER_PAGE
+    setCurrentSlide(nextSlide)
+  }
 
-    const autoplayInterval = autoplaySpeed
-      ? typeof autoplaySpeed === 'string'
-        ? parseFloat(autoplaySpeed)
-        : autoplaySpeed
-      : 0
+  const autoplayInterval = autoplaySpeed
+    ? typeof autoplaySpeed === 'string'
+      ? parseFloat(autoplaySpeed)
+      : autoplaySpeed
+    : 0
 
-    return (
-      <SliderContainer
-        autoplay={autoplay && autoplayInterval > 0}
-        autoplayInterval={autoplayInterval * 1000}
-        pauseOnHover
-        onNextSlide={this.handleNextSlide}
-        className={styles.container}
+  return (
+    <SliderContainer
+      autoplay={autoplay && autoplayInterval > 0}
+      autoplayInterval={autoplayInterval * 1000}
+      pauseOnHover
+      onNextSlide={handleNextSlide}
+      className={styles.container}
+    >
+      <Slider
+        loop
+        classes={{
+          root: handles.sliderRoot,
+          sliderFrame: handles.sliderFrame,
+        }}
+        perPage={PER_PAGE}
+        arrowRender={showArrows && ArrowRender}
+        currentSlide={currentSlide}
+        onChangeSlide={setCurrentSlide}
+        arrowsContainerComponent={showArrows && ArrowContainerRender}
+        duration={500}
       >
-        <Slider
+        {banners.map((banner, i) => (
+          <Slide
+            className={handles.slide}
+            key={i}
+            style={{ maxHeight: height }}
+            sliderTransitionDuration={500}
+          >
+            <Banner height={height} {...banner} />
+          </Slide>
+        ))}
+      </Slider>
+      {showDots && (
+        <Dots
           loop
+          perPage={PER_PAGE}
+          currentSlide={currentSlide}
+          totalSlides={banners.length}
```

Comment:
Why not save `banners.length` into a variable right at the start of the component code?
I see you're using this at least three times. It might also improve the readability of the `if` statement to check if `banners.length === 0`

Reply:
I don't see as an improvement of readability, because in this case, the best name for a variable like this would be `bannersLength`.

Reply:
You have a point hahahah